client-library/target/generated-sources/com/service/clientlibrary/implementation/TodoItemsServiceImpl.java

```diff
@@ -0,0 +1,270 @@
+        }
+        return (Response<BinaryData>) response;
+    }
+
+    @Override
+    public Response<BinaryData> updateSync(String endpoint, String contentType, long id,
+            String accept, BinaryData patch, RequestOptions requestOptions) {
+        return updateSync(defaultPipeline, endpoint, contentType, id, accept, patch, requestOptions);
+    }
+
+    private Response<BinaryData> updateSync(HttpPipeline pipeline, String endpoint,
+            String contentType, long id, String accept, BinaryData patch,
+            RequestOptions requestOptions) {
+        String host = endpoint + "/items/" + id;
+
+        // create the request
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.PATCH, host);
+
+        // set the headers
+        HttpHeaders headers = new HttpHeaders();
+        headers.add(HttpHeaderName.fromString("content-type"), String.valueOf(contentType));
+        headers.add(HttpHeaderName.fromString("accept"), String.valueOf(accept));
+        httpRequest.setHeaders(headers);
+
+        // add RequestOptions to the request
+        httpRequest.setRequestOptions(requestOptions);
+
+        // set the body content if present
+        httpRequest.setBody(BinaryData.fromObject(patch, serializer));
+
+        // send the request through the pipeline
+        Response<?> response = pipeline.send(httpRequest);
+
+        final int responseCode = response.getStatusCode();
+        boolean expectedResponse = responseCode == 200;
+        if (!expectedResponse) {
+            throw new RuntimeException("Unexpected response code: " + responseCode);
+        }
+        ResponseBodyMode responseBodyMode = null;
+        if (requestOptions != null) {
+            responseBodyMode = requestOptions.getResponseBodyMode();
+        }
+        if (responseBodyMode == ResponseBodyMode.DESERIALIZE) {
+            BinaryData responseBody = response.getBody();
+            HttpResponseAccessHelper.setValue((HttpResponse<?>) response, responseBody);
+        } else {
+            BinaryData responseBody = response.getBody();
+            HttpResponseAccessHelper.setBodyDeserializer((HttpResponse<?>) response, (body) -> responseBody);
+        }
+        return (Response<BinaryData>) response;
+    }
+
+    @Override
+    public Response<Void> deleteSync(String endpoint, long id, String accept,
+            RequestOptions requestOptions) {
+        return deleteSync(defaultPipeline, endpoint, id, accept, requestOptions);
+    }
+
+    private Response<Void> deleteSync(HttpPipeline pipeline, String endpoint, long id,
+            String accept, RequestOptions requestOptions) {
+        String host = endpoint + "/items/" + id;
+
+        // create the request
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.DELETE, host);
+
+        // set the headers
+        HttpHeaders headers = new HttpHeaders();
+        headers.add(HttpHeaderName.fromString("accept"), String.valueOf(accept));
```

Comment:
Not related to this PR at all, more thoughts I have that I'm just jotting down.

I wonder if we should look into having multiple internal implementations of `HttpHeaders` where small header sets use different patterns. For example, this is just a single header value but we instantiate the backing `Map` and all the overhead that goes along with that. Could we change this to a pattern where we defer `Map` creation until we seen 4 or 5 different header values?